src/abstractdataframe/iteration.jl

```diff
@@ -31,7 +31,7 @@ Base.size(itr::DataFrameRows) = (size(itr.df, 1), )
 Base.IndexStyle(::Type{<:DataFrameRows}) = Base.IndexLinear()
 @inline function Base.getindex(itr::DataFrameRows, i::Int)
     @boundscheck checkbounds(itr, i)
-    return DataFrameRow(itr.df, i)
+    return DataFrameRow(itr.df, i, :)
```

Comment:
I have it now in deprecated.jl, but I can move it back.

The reason why I did not allow it is that I do the same for `SubDataFrame` (i.e. deprecate two-argument version) because there is a confusion if the index means a row or a column (as for an `AbstractDataFrame` normally single index means a column).

I assume that this constructor will not be used directly in user's code, as it is shorter to write `df[i, :]`, but I wanted to have an explicit call to the constructor here so that it is clearer in the code what we do.

In summary - I would leave the deprecation, because directly calling `DataFrameRow` constructor is not something that is needed in user's code.

Reply:
Do you agree to deprecate two-argument versions?